lightning/src/chain/channelmonitor.rs

```diff
@@ -1127,6 +1127,29 @@ macro_rules! _process_events_body {
 }
 pub(super) use _process_events_body as process_events_body;
 
+pub(super) struct WithChannelMonitor<'a, 'b, L: Deref, Signer: WriteableEcdsaChannelSigner> where L::Target: Logger {
+	pub logger: &'a L,
+	pub channel_monitor: &'b ChannelMonitor<Signer>
+}
+
+impl<'a, 'b, L: Deref, Signer: WriteableEcdsaChannelSigner> Logger for WithChannelMonitor<'a, 'b, L, Signer> where L::Target: Logger {
+	fn log(&self, record: &Record) { self.logger.log(record); }
+	fn peer_id(&self) -> Option<PublicKey> { self.channel_monitor.get_counterparty_node_id() }
+	fn channel_id(&self) -> Option<[u8; 32]> { Some(self.channel_monitor.get_funding_txo().0.to_channel_id()) }
+}
+
+pub(super) struct WithChannelMonitorImpl<'a, 'b, L: Deref, Signer: WriteableEcdsaChannelSigner> where L::Target: Logger {
+	pub logger: &'a L,
+	pub channel_monitor_impl: &'b ChannelMonitorImpl<Signer>
+}
```

Comment:
played around with it abit, i think this isnt possible (not 100% sure, my rust isnt good)

the problem is the ChannelMonitorImpl methods, since ChannelMonitorImpl is self, passing that in requires it to not be `&mut self`. it ends up being something like this:
`WithChannelMonitor { logger, channel_monitor: &ChannelMonitor::from_impl(*self) }`
but this only works with methods that arent `&mut self`. 

the other side probably works abit more correctly, with something like this.
```
let binding = monitor.inner.lock().unwrap();
let logger = WithChannelMonitorImpl { logger: &self.logger, channel_monitor_impl: binding.deref() };
```

Reply:
> played around with it abit, i think this isnt possible (not 100% sure, my rust isnt good)
> 
> the problem is the ChannelMonitorImpl methods, since ChannelMonitorImpl is self, passing that in requires it to not be `&mut self`. it ends up being something like this: `WithChannelMonitor { logger, channel_monitor: &ChannelMonitor::from_impl(*self) }` but this only works with methods that arent `&mut self`.
> 

Hmm... I don't think we want to go down that path as `*self` is a move, which I don't think Rust would permit nor would we find desirable.

> the other side probably works abit more correctly, with something like this.
> 
> ```
> let binding = monitor.inner.lock().unwrap();
> let logger = WithChannelMonitorImpl { logger: &self.logger, channel_monitor_impl: binding.deref() };
> ```

Yeah, that would be better with a preference to calling the variable `inner` instead of `binding`. It may take some work to placate the compiler though. If you can't get it to work, I *think* we can at least use a local variable for lthe logger inside the `ChannelMonitorImpl` methods by having `WithChannelMonitorImpl` hold a reference to a mutable reference (i.e., `&'b &'b mut ChannelMonitorImpl`). That way we don't need to repeat the wrapping at each logging call site.